language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -85,19 +87,19 @@ main = do
     opts@Options{..} <- execParser optionsParserInfo
     ps <- readPackages optInputDars
     case mergePackageMap ps of
-      Left err -> fail err
+      Left err -> fail . T.unpack $ err
       Right pm ->
         forM_ (Map.toList pm) $
         \(pkgId, (mbPkgName, pkg)) -> do
           let id = show $ unPackageId pkgId
-          let name = fromMaybe id mbPkgName
-          let asName = if name == id then "itself" else name
+              name = maybe id (T.unpack . unPackageName) mbPkgName
```

Comment:
Why did you remove the `let`s? OCaml's `let x = ... in let y = ... in ...` is superior to Haskell's `let { x = ...; y = ... } in ...`, which is basically a huge footgun.

Reply:
We had,
```
  let x = ...
  let y = ...
  ...
```
in a monadic context. I replaced with,
```
let x = ...
     y = ...
    ....
```
as I believed it to be more idiomatic? You want I restore it?

Reply:
Let me know if so and I'll do so in a follow up PR!

Reply:
I don't know which one is more idiomatic. Let's go with the safer version then, i.e., the old version.